src/app/containers/App/App.jsx

```diff
@@ -6,22 +6,35 @@
 import { Component } from 'react';
 import { renderRoutes } from 'react-router-config';
 import { withRouter } from 'react-router-dom';
+import getRouteProps from '../../routes/getInitialData/utils/getRouteProps';
 
 import loadInitialData from '../../routes/loadInitialData';
 
 export class App extends Component {
   constructor(props) {
     super(props);
 
+    const { service, isAmp } = getRouteProps(
+      this.props.routes,
+      this.props.location.pathname,
+    );
+
     this.state = {
       data: this.props.initialData,
+      service: service,
+      isAmp: isAmp,
       loading: false,
       error: null,
       loadInitialDataPromise: null,
-    };  
+    };
   }
 
   async componentDidUpdate({ location: prevLocation }) {
+    const { service, isAmp } = getRouteProps(
```

Comment:
Not the keenest of this, as any time the component updates we run `getRouteProps` regardless of if we need to or not :/ Not sure of a nice alternative though

Reply:
This could be changed in a new refactor, which there is already an issue for.